Model/Api/Data/AutomatedTesting/Order.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2021 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api\Data\AutomatedTesting;
+
+class Order implements \JsonSerializable
+{
+    /**
+     * @var Address
+     */
+    private $billingAddress;
+
+    /**
+     * @var Address
+     */
+    private $shippingAddress;
+
+    /**
+     * @var string
+     */
+    private $shippingMethod;
+
+    /**
+     * @var string
+     */
+    private $shippingAmount;
+
+    /**
+     * @var OrderItem[]
+     */
+    private $orderItems;
+
+    /**
+     * @var string
+     */
+    private $subtotal;
+
+    /**
+     * @var string
+     */
+    private $tax;
+
+    /**
+     * @var string
+     */
+    private $discount;
+
+    /**
+     * @var string
+     */
+    private $grandTotal;
+
+    /**
+     * @return Address
+     */
+    public function getBillingAddress()
+    {
+        return $this->billingAddress;
+    }
+
+    /**
+     * @return Address
+     */
+    public function getShippingAddress()
+    {
+        return $this->shippingAddress;
+    }
+
+    /**
+     * @return string
+     */
+    public function getShippingMethod() {
+        return $this->shippingMethod;
+    }
+
+    /**
+     * @return string
+     */
+    public function getShippingAmount()
+    {
+        return $this->shippingAmount;
+    }
+
+    /**
+     * @return OrderItem[]
+     */
+    public function getOrderItems()
+    {
+        return $this->orderItems;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSubtotal()
+    {
+        return $this->subtotal;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTax()
+    {
+        return $this->tax;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDiscount()
+    {
```

Comment:
Do we consider having few discounts here?

Reply:
not at this stage. this amount currently is an aggregate of all the discounts applied. i would have to do a lookup into salesrules table to fetch a particular coupon. this can be added laters